F1WM/Repositories/NewsRepository.cs

```diff
@@ -33,7 +34,6 @@ public async Task<IEnumerable<NewsSummary>> GetLatestNews(int count, int? firstI
 					.Include(n => n.Topic)
 					.Where(n => !n.NewsHidden)
 					.OrderByDescending(n => n.Date)
-					.Take(count)
```

Comment:
This is effectively loading **all news ever** into memory. Please change it to IQueryable and remove ToListAsync to prevent evaluation of the query at this point.